### ISSUE TITLE
bug: don't consider "extensions" for directories when sorting

### ISSUE THREAD

**Issue by sdht0** [eza-community/eza#821]
```bash
$ mkdir a b.m c.n d.m e ; touch f.txt g.png h.png

$ eza --group-directories-first                              
a  b.m  c.n  d.m  e  f.txt  g.png  h.png

$ eza --group-directories-first --sort=extension
a  e  b.m  d.m  c.n  g.png  h.png  f.txt
```

The directories are also being sorted by extension based on the dot in the names. even though directories don't have extensions.

**Comment by cafkafk** [comment]
I don't think this technically classifies as a bug (as it is technically correct), but I'd love to see more input on whether this should be changed to reflect your desired behavior. Tentatively, I do agree that this should be changed, but I haven't thought much about it.

**Comment by sdht0** [comment]
One way to think about it is that directories have an implicit `.dir` extension and thus any other `.` in the directory name should be ignored.

I've found the current behavior surprising. For instance, when listing the installed packages on NixOS, I'd expect the multiple `eza` versions to be listed together. However, `eza-0.17.3` and `eza-0.18.0` are listed based on the `.3` and `.0`, which is not what I want. On the other hand, all files sorted by their extensions is what I want.

Given [xkcd 1172](https://xkcd.com/1172),  maybe `eza` can offer a new `--sort=fileextension` (or equivalent) option?

**Comment by Omnikron13** [comment]
![no-sort-dir](https://github.com/eza-community/eza/assets/2133262/8c7bf2bb-1641-40ff-96be-c84488b29382)

This is how it behaves if the sorting code just short circuits out to 'equal' if either the A or B input is a directory when it is asked to sort by extension.

**Comment by Omnikron13** [comment]
[Crude fork that fixes this.](https://github.com/Omnikron13/eza/tree/no-dir-ext)

**Comment by sdht0** [comment]
I am hoping this will be considered more seriously in the near term. Just listed a directory containing subdirectories with names like  `01. people`, `02. cities`, etc and `eza --group-directories-first --color=auto --sort=extension` sorted 02 before 01.

**Comment by sdht0** [comment]
IMO, using `--sort=extension` should probably imply `--group-directories-first` and directories should be sorted by names only.

Alternative is to think of directories as containing an implicit `.directory` extension and be sorted according (e.g., all directories are sorted to be after file.csv and before ventoy.exe). Not horrible I'd say, but opinions can clearly differ as above.

@cafkafk will you accept a patch for this?

**Comment by sdht0** [comment]
Thinking more on this, the simplest solution is to treat all directories exactly as if no `.` is present in the name (i.e., `None` extension). 

This was bothering me enough to implement a fix. Thanks @Omnikron13 for showing how to do this.

**Comment by cafkafk** [comment]
Fixed on main via 4f1ba6f7 — directories are now excluded from extension-based sorting. This hasn't shipped in a release yet (latest tag is v0.23.4, from before the fix landed), so you'll see the corrected behavior starting with the next release. Closing since the fix is merged.
